core/src/main/java/kafka/server/share/SharePartition.java

```diff
@@ -1321,7 +1321,111 @@ private AcquisitionLockTimerTask acquisitionLockTimerTask(
+            } else {
+                List<PersisterStateBatch> stateBatches = new ArrayList<>();
+                NavigableMap<Long, InFlightBatch> subMap = cachedState.subMap(floorOffset.getKey(), true, lastOffset, true);
+                for (Map.Entry<Long, InFlightBatch> entry : subMap.entrySet()) {
+                    InFlightBatch inFlightBatch = entry.getValue();
+
+                    if (inFlightBatch.offsetState() == null
+                            && inFlightBatch.batchState() == RecordState.ACQUIRED
+                            && checkForStartOffsetWithinBatch(inFlightBatch.firstOffset(), inFlightBatch.lastOffset())) {
+                        // For the case when batch.firstOffset < start offset <= batch.lastOffset, we will be having some
+                        // acquired records that need to move to archived state despite their delivery count.
+                        inFlightBatch.maybeInitializeOffsetStateUpdate();
+                    }
+
+                    // Case when the state of complete batch is valid
+                    if (inFlightBatch.offsetState() == null) {
+                        releaseAcquisitionLockOnTimeoutForCompleteBatch(inFlightBatch, stateBatches, memberId);
+                    } else { // Case when batch has a valid offset state map.
+                        releaseAcquisitionLockOnTimeoutForPerOffsetBatch(inFlightBatch, stateBatches, memberId, firstOffset, lastOffset);
+                    }
+                }
+                if (!stateBatches.isEmpty()) {
+                    // Even if write share group state RPC call fails, we will still go ahead with the state transition.
+                    isWriteShareGroupStateSuccessful(stateBatches);
+                }
+                // Update the cached state and start and end offsets after releasing the acquisition lock on timeout.
+                maybeUpdateCachedStateAndOffsets();
+            }
+        } finally {
+            lock.writeLock().unlock();
+        }
+    }
+
+    private void releaseAcquisitionLockOnTimeoutForCompleteBatch(InFlightBatch inFlightBatch,
+                                                                 List<PersisterStateBatch> stateBatches,
+                                                                 String memberId) {
+        if (inFlightBatch.batchState() == RecordState.ACQUIRED) {
+            InFlightState updateResult = inFlightBatch.tryUpdateBatchState(
+                    inFlightBatch.lastOffset() < startOffset ? RecordState.ARCHIVED : RecordState.AVAILABLE,
+                    false,
+                    maxDeliveryCount,
+                    EMPTY_MEMBER_ID);
+            if (updateResult == null) {
+                log.debug("Unable to release acquisition lock on timeout for the batch: {}"
+                        + " for the share partition: {}-{}-{}", inFlightBatch, groupId, memberId, topicIdPartition);
```

Comment:
I think the share partition is {groupId}-{topicIdPartition}. The member ID is something else. I'd change the formatting of the message to separate these concepts.